### ISSUE TITLE
`react/no-unused-prop-types` when use props in nested function in functional component

### ISSUE THREAD

**Issue by AlphaBeth** [yannickcr/eslint-plugin-react#1234]
This code
```jsx
import React from 'react';
import PropTypes from 'prop-types';

function TestComponent(props) {
    function getA() {
        return props.a;
    }
    function getB() {
        return props.b;
    }
    function getCD() {
        return props.c + props.d;
    }
    return (<div>
        {getA()}
        {getB()}
        {getCD()}
    </div>);
}

TestComponent.propTypes = {
    a: PropTypes.string.isRequired,
    b: PropTypes.string.isRequired,
    c: PropTypes.string.isRequired,
    d: PropTypes.string.isRequired,
};

export default TestComponent;
```
will raise errors
```
  22:8  error  'a' PropType is defined but prop is never used  react/no-unused-prop-types
  23:8  error  'b' PropType is defined but prop is never used  react/no-unused-prop-types
```
for all props except used in last function.

**Comment by ljharb** [comment]
This should certainly be fixed; but you may want instead to destructure the props in the signature, which would avoid this problem.

**Comment by jseminck** [comment]
This was fixed by: https://github.com/yannickcr/eslint-plugin-react/pull/1218

To confirm, I ran your exact test-case on master and it now passes as valid!
